app/metal-controller-manager/controllers/server_controller.go

```diff
@@ -61,24 +68,26 @@ func (r *ServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{RequeueAfter: constants.DefaultRequeueAfter}, err
 	}
 
-	f := func(ready, requeue bool) (ctrl.Result, error) {
-		s.Status.Power = "off"
+	s.Status.Power = "off"
 
-		poweredOn, err := mgmtClient.IsPoweredOn()
-		if err != nil {
-			s.Status.Power = "unknown"
-		}
+	poweredOn, powerErr := mgmtClient.IsPoweredOn()
```

Comment:
I understand that power state was read in `defer` to make sure we read the status after we do any changes, but looks like reconcile loop is anyway launched once more, so next interation reads latest state, but this change allows to read the power state only once for the reconcile loop